[Woo POS] Coupons: Preparation work for pagination

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Preparation work for WOOMOB-129

## Description
This PR starts the work on coupons pagination by updating the service to account for the existence of paged items, as well as the different coupons controller functions, to take into account that pagination might exist.

No real functional changes have been made and we're still fetching only the 1st page from remote, the only difference is that we perform calls to `loadNextItems()` now when we scroll, which we didn't before.

## Changes
- Updated `loadFirstPage()` to be `fetchCoupons(pageNumber: n)`, we reuse this function for all item protocol calls, the only thing that changes is the page passed through.
- Updated the coupon service signatures to return `PagedItems<POSItem>` instead of a simple array, as we'll need to know if has more items in order to update the state
- Updated `setCouponsLoadedViewState` to accept a `hasMoreItems` property, this is used to set the view state and check if we need to trigger a `loadNextItems()` from `ItemList`. At the moment, we trigger it every time we scroll.

## Testing information
- Enable `.enableCouponsInPointOfSale`
- Add a break point in `await fetchCoupons(pageNumber: pageNumber)` in `PointOfSaleCouponsController` line 43
- Load POS and go to coupons
- You should see no difference in expected initial loading behavior
- Scroll down, observe how the breakpoint is triggered once the `ItemList` triggers infinite scroll.

As mentioned the work here is just preparatory, there's no proper pagination implemented yet, and we'll continue triggering a page 1 load when the infinite scrolling trigger is called.

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
pass pageNumber prop when fetch coupons

iamgabrielma committed:
extract view state to helpers

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
Update state to assure loadNextItems is triggered

We’ll pass a property later on based on if there’s more items to load, for now let’s assure that loadNextItems is called when we check view state

iamgabrielma committed:
update service signatures to return PagedItems<POSItem> type

staskus reviewed (approved):
👍 Nice! Let's continue 🚀
